source/agora/common/DNS.d

```diff
@@ -585,6 +584,75 @@ public struct ResourceRecord
     }
 }
 
+/// The OPT opcode is a special ResourceRecord with its own semantic
+/// See https://datatracker.ietf.org/doc/html/rfc6891
+public struct OPTRR
+{
+    @safe pure nothrow @nogc:
+
+    /***************************************************************************
+
+        The underlying ResourceRecord
+
+        OPT records are backed by RR, but their fields are to be interpreted
+        differently, e.g. `ttl` is a bitfield (since OPT MUST NOT be cached),
+        and `CLASS` is the UDP payload size.
+
+        This struct wraps the record and expose a set of get / set properties
+        for easy manipulation.
+
+        See_Also:
+          https://datatracker.ietf.org/doc/html/rfc6891#section-6.1.3
+
+    ***************************************************************************/
+
+    public ResourceRecord record = ResourceRecord(Domain.init, TYPE.OPT, cast(CLASS) 4096);
+
+    /// Requestor's UDP payload size
+    public ushort payloadSize () scope const
+    {
+        return this.record.class_;
+    }
+
+    /// Ditto
+    public void payloadSize (ushort value) scope
+    {
+        this.record.class_ = cast(CLASS) value;
+    }
+
+    /// Upper byte of the header RCODE
+    /// Currently only BADVERS (0x01) is supported
+    public ubyte extendedRCODE () scope const
+    {
+        return this.record.ttl >> 24;
+    }
+
+    /// Ditto
+    public void extendedRCODE (ubyte upperByteValue) scope
+    {
+        this.record.ttl = (this.record.ttl & 0x00FF_FFFF) | upperByteValue;
```

Comment:
Shifting required
`this.record.ttl = (this.record.ttl & 0x00FF_FFFF) | (upperByteValue << 24);`

Reply:
Good catch

Reply:
Rebased, fixed.